InnerEye/ML/Histopathology/datamodules/base_module.py

```diff
@@ -46,16 +50,25 @@ def __init__(self, root_path: Path, max_bag_size: int = 0, batch_size: int = 1,
         :param cache_mode: The type of caching to perform, i.e. whether the results of all
         transforms up to the first randomised one should be computed only once and reused in
         subsequent iterations:
-          - `MEMORY`: the entire transformed dataset is kept in memory for fastest access;
-          - `DISK`: each transformed sample is saved to disk and loaded on-demand;
-          - `NONE` (default): no caching is performed.
-        :param save_precache: Whether to pre-cache the entire transformed dataset upfront and save
+          - `MEMORY`: MONAI CacheDataset is used, the entire transformed dataset is kept in memory for fastest access;
+          - `DISK`: MONAI PersistentDataset is used, each transformed sample is saved to disk and loaded on-demand;
+          - `NONE` (default): standard MONAI dataset is used, no caching is performed.
+        :param precache_location: Whether to pre-cache the entire transformed dataset upfront and save
         it to disk. This is done once in `prepare_data()` only on the local rank-0 process, so
-        multiple processes can afterwards access the same cache without contention in DDP settings.
+        multiple processes can afterwards access the same cache without contention in DDP settings. 3 options are
+        available:
+          - `CPU`: each transformed sample is saved to disk and loaded into CPU memory on-demand;
+          - `GPU`: each transformed sample is saved to disk and loaded into GPU memory on-demand;
+          - `NONE (default)`: no pre-cache is performed;
         :param cache_dir: The directory onto which to cache data if caching is enabled.
         :param number_of_cross_validation_splits: Number of folds to perform.
         :param cross_validation_split_index: Index of the cross validation split to be performed.
         """
+        if precache_location == CacheLocation.NONE:
+            save_precache = None
+        else:
+            save_precache = precache_location
```

Comment:
I was trying to keep low the number of parameters relative to cache to avoid further confusion between the different options. the only reason why here I redefined a save_precache was to make the assert statement a bit more compact. Let me remove, I see this confused both you and Anton

Reply:
save_precache removed